Little fixes + Match ended text

2duviz2 opened the pull request:
This pull adds a big "match ended" text when you finish a match, its simple but necessary to avoid confusing the players. It also hides the currency UI so when you extract, the opponent's text isn't hidden by your current money that you wont use.

2duviz2 committed:
Match ended text

2duviz2 committed:
Repo font

It makes it so you dont have to click on last match results to show the new font

2duviz2 committed:
Update DanosMainMenuManager.cs

2duviz2 committed:
Merge remote-tracking branch 'upstream/main' into visuals-revamp

havaka commented:
it'd be really great if the post-game screen showed match data like broken money, extracted money, match duration, maybe even valuables map. i'm not really good at c#, but afaik you can use libraries to get this data.
for example, [LevelScaling mod](https://thunderstore.io/c/repo/p/Swaggies/LevelScaling/source/) (last lines):
```csharp
if (SemiFunc.RunIsLevel())
{
  TextMeshProUGUI obj7 = statsText;
  ((TMP_Text)obj7).text = ((TMP_Text)obj7).text + "ORB DROPS\n";
  string text3 = Helpers.ColoredText(LevelStatsManager.enemyKills.ToString(), LevelStatsManager.allOrbsDropped ? colorGood : ((LevelStatsManager.enemyKills == 0) ? colorNone : colorBase));
  TextMeshProUGUI obj8 = statsNumbers;
  ((TMP_Text)obj8).text = ((TMP_Text)obj8).text + text3 + "\n";
  TextMeshProUGUI obj9 = statsText;
  ((TMP_Text)obj9).text = ((TMP_Text)obj9).text + "EXTRACTED\n";
  string text4 = Helpers.ColoredText("$" + SemiFunc.DollarGetString((int)LevelStatsManager.totalValueExtracted), (!(LevelStatsManager.totalValueExtracted < (float)LevelStatsManager.totalMapValue)) ? colorGood : ((LevelStatsManager.totalValueExtracted < 1f) ? colorNone : colorBase));
  string text5 = Helpers.ColoredText("/$" + SemiFunc.DollarGetString(LevelStatsManager.totalMapValue), colorSubtext);
  TextMeshProUGUI val = statsNumbers;
  ((TMP_Text)val).text = ((TMP_Text)val).text + text4 + " <size=75%>" + text5 + "</size>\n";
  TextMeshProUGUI obj10 = statsText;
  ((TMP_Text)obj10).text = ((TMP_Text)obj10).text + "BROKEN\n";
  string text6 = Helpers.ColoredText("-$" + SemiFunc.DollarGetString((int)LevelStatsManager.totalValueBroken), (LevelStatsManager.totalValueBroken < 1f) ? colorNone : colorBad);
  TextMeshProUGUI obj11 = statsNumbers;
  ((TMP_Text)obj11).text = ((TMP_Text)obj11).text + text6 + "\n";
  TextMeshProUGUI obj12 = statsText;
  ((TMP_Text)obj12).text = ((TMP_Text)obj12).text + "FOUND\n";
  string arg = Helpers.ColoredText($"/{LevelStatsManager.valuablesTotal}", colorSubtext);
  string text7 = Helpers.ColoredText($"{LevelStatsManager.valuablesFound} <size=75%>{arg}</size>", (LevelStatsManager.valuablesFound >= LevelStatsManager.valuablesTotal) ? colorGood : ((LevelStatsManager.valuablesFound < 1) ? colorNone : colorBase));
  TextMeshProUGUI obj13 = statsNumbers;
  ((TMP_Text)obj13).text = ((TMP_Text)obj13).text + text7 + "\n";
  TextMeshProUGUI obj14 = statsText;
  ((TMP_Text)obj14).text = ((TMP_Text)obj14).text + "EXPLORED\n";
  string arg2 = Helpers.ColoredText($"/{LevelStatsManager.totalModules}", colorSubtext);
  string text8 = Helpers.ColoredText($"{LevelStatsManager.modulesExplored} <size=75%>{arg2}</size>", (LevelStatsManager.modulesExplored >= LevelStatsManager.totalModules) ? colorGood : ((LevelStatsManager.modulesExplored < 1) ? colorNone : colorBase));
  TextMeshProUGUI obj15 = statsNumbers;
  ((TMP_Text)obj15).text = ((TMP_Text)obj15).text + text8 + "\n";
}
```

2duviz2 commented:
Okay I'll go for that! I'll make the request draft while I do it

2duviz2 committed:
Time in match popup

Changes the post-match popup to show the time you have been on the match

2duviz2 commented:
Okay at this moment it would be really hard to keep track of broken objects since I'm not used to the code of the mod yet, maybe in other moment, so the pull is ready

2duviz2 committed:
Added a 1s cooldown before finishing the match because of death

This is just necessary to have some fun for clips, it just waits one second before telling the game that you died

2duviz2 commented:
Okay this should be all, I ran out of ideas

2duviz2 committed:
Added libraries to gitignore

Common name for references, avoids pushing random dlls

2duviz2 committed:
Fixed buttons overlapping when closing menus

I don't know why, but because of the submenus this project adds, they overlap even when not active, so the best solution is to straight up delete them. Also added real-time characters limit to the passphrase

2duviz2 committed:
Changed round ended text a bit
